shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/common/extractor/SQLHintExtractor.java

```diff
@@ -17,54 +17,88 @@
+    
+    private final Set<String> supportedSet = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
+    
+    public SQLHintExtractor(final SQLStatement sqlStatement) {
+        supportedSet.add(SQL_HINT_DATASOURCE_NAME_KEY);
+        supportedSet.add(SQL_HINT_WRITE_ROUTE_ONLY_KEY);
+        if (sqlStatement instanceof AbstractSQLStatement) {
+            sqlHintMap = extract((AbstractSQLStatement) sqlStatement);
+        } else {
+            sqlHintMap = Collections.emptyMap();
+        }
+    }
     
     /**
-     * Find hint data source name.
+     * Extract from statement.
      *
      * @param statement statement
-     * @return data source name
+     * @return sql hint map
      */
-    public static Optional<String> findHintDataSourceName(final AbstractSQLStatement statement) {
+    public Map<String, String> extract(final AbstractSQLStatement statement) {
+        Map<String, String> result = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
         for (CommentSegment each : statement.getCommentSegments()) {
-            Optional<String> result = findDataSourceNameFromComment(each.getText());
-            if (result.isPresent()) {
-                return result;
-            }
+            extractFromComment(each.getText(), result);
         }
-        return Optional.empty();
+        return result;
     }
     
-    private static Optional<String> findDataSourceNameFromComment(final String comment) {
+    private void extractFromComment(final String comment, final Map<String, String> result) {
         int startIndex = comment.toLowerCase().indexOf(SQL_HINT_TOKEN);
         if (startIndex < 0) {
-            return Optional.empty();
+            return;
         }
         startIndex = startIndex + SQL_HINT_TOKEN.length();
         int endIndex = comment.endsWith(SQL_COMMENT_SUFFIX) ? comment.indexOf(SQL_COMMENT_SUFFIX) : comment.length();
         String[] hintValue = comment.substring(startIndex, endIndex).trim().split(SQL_HINT_SPLIT);
-        if (2 == hintValue.length && SQL_HINT_DATASOURCE_NAME_KEY.equalsIgnoreCase(hintValue[0].trim()) && hintValue[1].trim().length() > 0) {
-            return Optional.of(hintValue[1].trim());
+        if (2 == hintValue.length && hintValue[0].trim().length() > 0 && hintValue[1].trim().length() > 0 && supportedSet.contains(hintValue[0].trim())) {
+            result.put(hintValue[0].trim(), hintValue[1].trim());
         }
-        return Optional.empty();
+    }
+    
+    /**
+     * Find hint data source name.
+     *
+     * @return data source name
+     */
+    public Optional<String> findHintDataSourceName() {
+        return Optional.ofNullable(sqlHintMap.get(SQL_HINT_DATASOURCE_NAME_KEY));
+    }
+    
+    /**
+     * Judge whether is hint routed to write data source.
+     *
+     * @return whether is hint routed to write data source
+     */
+    public boolean isHintWriteRouteOnly() {
+        return "true".equalsIgnoreCase(sqlHintMap.get(SQL_HINT_WRITE_ROUTE_ONLY_KEY));
```

Comment:
@tuichenchuxin Can we add `SQLHintProperties` and `SQLHintPropertyKey` to handle properties type cast?